src/Viewport.ts

```diff
@@ -0,0 +1,46 @@
+import { Point } from './Point'
+
+export class Viewport {
+  public center: Point
+  public zoomLevel: number
+  public gridUnit: number
+
+  constructor(zoomLevel: number, center?: Point){
+    this.zoomLevel = zoomLevel
+    this.center = center || new Point()
+    this.gridUnit = 10;
+  }
+
+  gridToPixel(point: Point): Point {
+    return new Point(
+      point.x * this.gridUnit,
+      point.y * this.gridUnit
+    )
+  }
+
+  pixelToGrid(point: Point): Point {
+    return new Point(
+      point.x / this.gridUnit,
+      point.y / this.gridUnit
+    )
+  }
+
+  zoomedPixelToGrid(point: Point): Point {
```

Comment:
I think this must become default `pixelToGrid` function

Reply:
My only worry is that we then have two functions with inverse name and they do different operations (because for gridToPixel the zoom is handled in the viewBox).